webapp/channels/src/components/channel_settings_modal/channel_access_rules_confirm_modal.tsx

```diff
@@ -143,9 +138,15 @@ function ChannelAccessRulesConfirmModal({
 
     const modalSubtitle = channelName;
 
+    // Dynamic tab content
     const currentUsers = activeTab === 'allowed' ? filteredAllowedUsers : filteredRestrictedUsers;
     const totalCount = activeTab === 'allowed' ? usersToAdd.length : usersToRemove.length;
 
+    // Don't show modal if there are no changes
+    if (!hasChanges) {
+        return null;
+    }
```

Comment:
Since we are no longer using `useMemo` for currentUsers and totalCount, we can bring back this to where it was.